### PR TITLE
AP-3864: Enable multiple opponents

### PR DESCRIPTION
**Pull request by colinbruce**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/AP-3864)

This PR updates the application -> opponent relationship from has_one to has_many
It then implements the `has-another` pattern for the application merits task opponents' name page

This required a re-write of the ccms/xml matchers to allow multiple values to be matched.  
Currently these only work in to assure all values are present, not their sequence, this may cause an issue in future with boolean values if they are dissimilar. e.g. expecting `[true, false]` it will ensure that one of each is present, but not necessarily that they are assigned to the expected linked blocks

Future work will expand the opponent to allow differentiation between individuals and organisations 

## Checklist

Before you ask people to review this PR:

- Tests and rubocop should be passing: `bundle exec rake`
- Github should not be reporting conflicts; you should have recently run `git rebase main`.
- There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- The PR description should say what you changed and why, with a link to the JIRA story.
- You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: app/helpers/task_list_helper.rb
Comment (naseberry) [inline app/helpers/task_list_helper.rb:22]: Complexity: I know our domain is complicated and this feels complex, is it be possible to make this less complex?

File: app/services/ccms/requestors/case_add_requestor.rb
Comment (naseberry) [inline app/services/ccms/requestors/case_add_requestor.rb:296]: Comment: Do we need to keep this comment?

File: app/services/ccms/requestors/case_add_requestor.rb
Comment (naseberry) [inline app/services/ccms/requestors/case_add_requestor.rb]: With limited knowledge of the payload, i may be talking squat, but as there can be multiple opponents, should this be an array?

**Review by naseberry — COMMENTED** [review]
I would require a second review from a ruby-ist.

PS - loving the GoT references in specs.

File: app/services/ccms/requestors/case_add_requestor.rb
Comment (colinbruce) [reply app/services/ccms/requestors/case_add_requestor.rb]: This looks like a rebase issue to me - I am going to return the PR to draft for now and address all of your -very valid- points!

Comment (colinbruce) [reply app/services/ccms/requestors/case_add_requestor.rb]: Yeah, a later commit had resolved this 😕 
Not sure how this was in the review state 😖

File: app/helpers/task_list_helper.rb
Comment (colinbruce) [reply app/helpers/task_list_helper.rb]: Have a look at the latest version @naseberry - happy to discuss

File: spec/requests/providers/application_merits_task/opponents_names_controller_spec.rb
Comment (kmahern) [inline spec/requests/providers/application_merits_task/opponents_names_controller_spec.rb:53]: Is this description correct? Should it be "displays opponent details"

File: app/services/flow/flows/provider_merits.rb
Comment (kmahern) [inline app/services/flow/flows/provider_merits.rb:64]: can the comment be removed?

File: spec/services/ccms/requestors/case_add_requestor_spec.rb
Comment (kmahern) [inline spec/services/ccms/requestors/case_add_requestor_spec.rb:221]: Is this correct? (that only the Sansa Opponent-Test is included  in the xml and not Joffrey Test-Opponent)? Could the description be clearer? (e.g. it "generates expected xml with first opponent")

Comment (colinbruce) [reply spec/services/ccms/requestors/case_add_requestor_spec.rb:221]: Good catch, it was my initial test that the _new_ value was at least found.  Extended to ensure it matches both :+1:

**Comment by colinbruce** [comment]
> I would require a second review from a ruby-ist.
> 
> PS - loving the GoT references in specs.

There were there to start, I just extended them! 😀

**Review by naseberry — APPROVED** [review]
lgtm

I would have preferred new views to use the govuk-components view helpers, however, we have tickets in the backlog to action this.

**Pull request merged**
